File: Strings/First Non-Repeating Character/O(n)_time_and_O(1)_space.py
def firstNonRepeatingCharacter(string):
    # Write your code here.
    charFreq = {}

    for char in string:
        charFreq[char] = charFreq.get(char,0) + 1

    for ind in range(len(string)):
        char=string[ind]
        if charFreq[char] == 1:
            return ind
            
    return -1

File: Strings/First Non-Repeating Character/O(n)_time_and_O(1)_space.js
function firstNonRepeatingCharacter(string) {
  // Write your code here.
  charCount = {};
  for (let char of string) {
    if (!(char in charCount)) charCount[char] = 0;
    charCount[char]++;
  }

  for (let ind = 0; ind < string.length; ind++) {
    const char = string[ind];
    if (charCount[char] == 1) return ind;
  }

  return -1;
}
